patches/eslint+8.54.0.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/eslint/lib/cli-engine/formatters/stylish.js b/node_modules/eslint/lib/cli-engine/formatters/stylish.js
+index a808448..1d4b79e 100644
+--- a/node_modules/eslint/lib/cli-engine/formatters/stylish.js
++++ b/node_modules/eslint/lib/cli-engine/formatters/stylish.js
+@@ -5,7 +5,7 @@
+ "use strict";
+ 
+ const chalk = require("chalk"),
+-    stripAnsi = require("strip-ansi"),
++    stripAnsi = require("strip-ansi-cjs"),
```

Comment:
this patch only applies within `eslint-mdx`, will downstream adopters run into an issue if they don't patch too?

Reply:
This is a package manager issue, nothing related to end user, I've described it in PR description.

Reply:
`yarn.lock` looks broken. For example:

```
"@types/acorn@^4.0.0", "@types/acorn@^6.0.0":
  version "6.0.0"
  resolved "https://registry.yarnpkg.com/@types/acorn/-/acorn-6.0.0.tgz#7ec9bfdad8e7db4cb03b2b5b05a6f13f77bbc874"
  integrity sha512-8tSIZoick397hAZIZkiq6R09qNgLjx0CBvmxix3gWyCi6UvXNFjN56xh4e1wTwyYlnTEy32tgo9cooxt6J4lhA==
  dependencies:
    "@types/estree" "*"
    acorn "*"
```

There is no overlap between semver ranges `^4.0.0` and `^6.0.0`. These should be two separate lockfile entries. This is just an example, there are more broken entries. This might be the root cause for several issues.

Maybe it’s best to split this PR into multiple smaller, more manageable ones:

- Update GitHub workflow versions
- Remove support for Node.js <18
- Fix/update `yarn.lock`
  - Switch to npm, since Yarn 1 is pretty much unmaintained
  - Get rid of those patches

Reply:
I'm using `resolutions` to lock the version for compatible with `unified/mdx` related types. There are `@types/acorn@4` because of non-upgraded `remark-lint` plugins mentioned at https://github.com/mdx-js/eslint-mdx/pull/486#issuecomment-1839062311.
So there is no action to do.